### DISCUSSION TITLE
x-mask:dynamic="$money($input)" doesn't work in this case where as in a standalone POC it does work

### DISCUSSION THREAD

**Discussion by anjanesh** [alpinejs/alpine#3806] [4. General]
`x-mask:dynamic="$money($input)"` works here :
```html
<input
    type="text"
    :value="
        $store.global_data['total'] = (
            parseFloat(String($store.global_data['x']).replace(',', '')) +
            parseFloat($store.global_data['a'].replace(',', '')) +
            parseFloat($store.global_data['b'].replace(',', '')) +
            parseFloat($store.global_data['c'].replace(',', '')) +
            parseFloat($store.global_data['d'].replace(',', '')) +
            parseFloat($store.global_data['e'].replace(',', ''))
        ).toFixed(2)
    "
    x-mask:dynamic="$money($input)"
    disabled
/>
```
But the commas are not displayed in this case :
```html
<input
    type="text" id="grand-total"
    :value="
        $store.global_data['grand_total'] = parseFloat(
            parseFloat(String($store.global_data['cloud_servers_total']).replace(',', '')) +
            parseFloat($store.global_data['managedServices_total'].replace(',', '')) +
            parseFloat($store.global_data['backup_total'].replace(',', '')) +
            parseFloat($store.global_data['disaster_total'].replace(',', '')) +
            parseFloat($store.global_data['transfer_total'].replace(',', '')) +
            parseFloat($store.global_data['others_total'].replace(',', ''))
        ).toFixed(2)
    "
    x-mask:dynamic="$money($input)"
    disabled
/>
```
If I remove `parseFloat(String($store.global_data['cloud_servers_total']).replace(',', ''))` it works (commas as displayed).

**Comment by ekwoka** [comment]
This is a mess.

Cns you clean this up?

Why are you doing so much float parsing and string replaces?

Why is this in  a single expression?

What does the data actually look like?

**Reply by anjanesh** [reply]
I have some 10 modals where each modal has a form with dynamic number of input rows with the total amount of money for each modal stored in Alpine.store.

I am having a disabled text input on the main page which should show the grand total which is the sum of all the modals' total value.

**Reply by ekwoka** [reply]
Okay, so why not have the store have a getter on it?

```js
{
   get grand_total() {
     return this.cloud_servers_total + this.managedServices_total + this.backup_total
   }
 }
 ```
 
 And if they are numbers, store them as numbers, not strings.
 
 There is no reason to put an assignment inside the value binding on the input element.

**Reply by anjanesh** [reply]
Thanks for the continued support to use getters and setters in JavaScript - I wasn't used this used to it at all.
I want to store as numbers but display commas in numbers - this is a most requested feature of all financial people.
For this partuclar case I have to let go off `x-mask:dynamic="$money($input)"` and return a formatted string of type `Intl.NumberFormat` and ad the same time save the number elsewhere in the getter method.

**Reply by ekwoka** [reply]
You don't need to save it anywhere.

You just use the getter.

Mask is meant for when the user inputs things.

It's not meant for you binding the value.

**Reply by anjanesh** [reply]
_You don't need to save it anywhere_ - I'm using `Alpine.store('global_data')` to store the data into `localStorage` which has the values as numbers where as `Alpine.store('local_data')` as formatted strings and other data that I don't want shared across pages.

**Reply by ekwoka** [reply]
Yeah, that's likely a poor way to do it

But you just use the getter still, you don't need to store it at all...it will work fine when you load it back from local storage regardless. It doesn't need to be saved.
